tests/unit/moduleapi/propagate.tcl

```diff
@@ -208,21 +208,33 @@ tags "modules" {
 
                     $master propagate-test.timer-maxmemory
 
+                    # The replica will have two keys: "notifications" and "timer-maxmemory-middle"
+                    # which are not volatile
+                    wait_for_condition 5000 10 {
+                        [$replica dbsize] eq 2
+                    } else {
+                        fail "Not all keys have been evicted"
+                    }
+
                     assert_replication_stream $repl {
                         {select *}
                         {multi}
                         {incr notifications}
                         {incr notifications}
-                        {set timer-maxmemory-start 1 PXAT *}
-                        {incr notifications}
-                        {del timer-maxmemory-start}
+                        {set timer-maxmemory-volatile-start 1 PXAT *}
                         {incr timer-maxmemory-middle}
                         {incr notifications}
-                        {set timer-maxmemory-end 1}
+                        {incr notifications}
+                        {set timer-maxmemory-volatile-end 1 PXAT *}
                         {exec}
+                        {incr notifications}
+                        {del timer-maxmemory-volatile-*}
+                        {incr notifications}
+                        {del timer-maxmemory-volatile-*}
```

Comment:
can you help me understand this change?
all i see that changed in the test code itself (and the module), is not sending the CONFIG SET maxmemory (which means the test name is now outdated?)
but i don't understand how it impacted the verification code in such a way.

p.s. is the reason you removed the CONFIG SET, just because you realized that MULTI-EXEC demonstrates a similar problem, so there's no need for the module notification case?

Reply:
in the past it used to do
Call(INCR) // and replicate
Call(CONFIG SET maxmemory 1)
Replicate(INCR) // so ti's done only on replica
Call(CONFIG SET maxmemory 0)
Call(INCR) //and replicate

after the last commit, setting maxmemory to 1 and then 0 in the same atomic operation would not result in eviction (because numops > 0). i wanted to get some evictions, so I removed the `CONFIG SET maxmemory 0` form the module code (note that it still does `CONFIG SET maxmemory 1` so the test name is correct)

another change is that I changed the names of the keys we INCR via RM_Call and made them volatile so that they can be evicted when I set the eviction to volatile-based (I don't use allkeys because we have the special key "notifications" that always comes back to life, and then it gets evicted a random amount of times, and I can't assert anything about the replication stream)

this test is indeed similar to the one in maxmemory.tcl but I wanted to test the module case too (also, the test already existed, would be a shame to delete it).

Reply:
now, we can see that keys are evicted only after the module executes and replicates everything in the callback